recipes/flann/all/conanfile.py

```diff
@@ -0,0 +1,116 @@
+import glob
+import os
+
+from conans import ConanFile, CMake, tools
+
+
+class LibFlannConan(ConanFile):
+    name = "flann"
+    description = "Fast Library for Approximate Nearest Neighbors"
+    topics = "conan", "flann"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.cs.ubc.ca/research/flann/"
+    license = "BSD-3-Clause"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake", "cmake_find_package"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_hdf5": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_hdf5": False
+    }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_hdf5:
+            self.requires("hdf5/1.10.6")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+        # Workaround issue with empty sources for a CMake target
+        flann_cpp_dir = os.path.join(self._source_subfolder, "src", "cpp")
+        with open(os.path.join(flann_cpp_dir, "empty.cpp"), "w") as fd:
```

Comment:
Replace with `tools.touch(os.path.join(flann_cpp_dir, "empty.cpp"))`

Reply:
Ooooh, I didn't know it existed, thanks :o

Reply:
Actually it fails if the file doesn't exist (as per the documentation), so I can't use `tools.touch` here.

Reply:
`tools.save("empty.cpp", "\n")` is your friend then

Reply:
There are so many functions in `tools` 🤔